ansible_collections/arista/avd/plugins/plugin_utils/schema/avdvalidator.py

```diff
@@ -109,6 +111,16 @@ def _keys_validator(validator, keys: dict, instance: dict, schema: dict):
         if "dynamic_valid_values" in childschema:
             childschema.setdefault("valid_values", []).extend(get_all(instance, childschema["dynamic_valid_values"]))
 
+        # Validation of passwords
+        if childschema.get("password_type") and childschema.get("password_key_field"):
+            try:
+                password_type = childschema.get("password_type")
+                password_key_field = childschema.get("password_key_field")
+                password_key = get_all(instance, password_key_field, required=True)[0]
+                yield from _password_validator(validator, password_type, password_key, instance, schema)
```

Comment:
I would prefer to move most of this logic to the validator.
instead of fetching out the password related keys here, I would just pass along the `instance` as `parent` and `childschema` as `schema` and `instance[key]` as `instance`.